### PR TITLE
Adds in the ability for reagent reactions while breathing

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Added the ability for the lungs to process reagent reactions, you can observe this by breathing nitrous oxide for a long period.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs [hunk 1]
```diff
@@ -397,6 +400,16 @@ public virtual void ToxinBreathinCheck(GasMix gasMix)
 			}
 		}
 
+		private void BreathInEffects(GasMix mix)
+		{
+			var reagents = mix.GetReagentMix();
+			foreach (var reaction in ReagentContainerBody.ReactionSet.reactions)
+			{
+				if (reaction.IsReactionValid(reagents) == false) continue;
+				reaction.ForceApply(reagents, LivingHealthMaster.gameObject);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs:409]: Sure you can just use the reagent mix to do this?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs:409]: actually wait no
no we cant

it needs a metabolism component, this skips the metabolism part and makes the lung responsible for checking and running the code for that

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs:409]: why?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs:409]: > it needs a metabolism component

**Comment by MaxIsJoe** [comment]
fixed in #10339

**Pull request closed without merging**